command.go

```diff
@@ -106,6 +108,7 @@ type baseCmd struct {
 	ctx  context.Context
 	args []interface{}
 	err  error
+	keyOffset int
```

Comment:
Using `keyBase+keyOffset` looks a bit complicated. Can't we rename this to `keyPos` and use it when `cmd.keyPos != 0` and fallback to `info.FirstKeyPos`?